Library/Homebrew/formula.rb

```diff
@@ -1318,18 +1318,25 @@ def verify_download_integrity(fn)
 
   # @private
   def run_test
-    old_home = ENV["HOME"]
-    build, self.build = self.build, Tab.for_formula(self)
-    mktemp do
-      @testpath = Pathname.pwd
-      ENV["HOME"] = @testpath
-      setup_home @testpath
-      test
+    preserve_files = ARGV.keep_files?
+    begin
+      old_home = ENV["HOME"]
+      build, self.build = self.build, Tab.for_formula(self)
+      mktemp("#{name}-", { :preserve => preserve_files }) do
+        @testpath = Pathname.pwd
+        ENV["HOME"] = @testpath
+        setup_home @testpath
+        test
+      end
+    ensure
+      if preserve_files
+        ohai "Kept temporary files"
+        puts "Temporary files are saved at #{@testpath}"
+      end
```

Comment:
Then make that: `ohai "Retained temporary files:", @testpath if ARGV.keep_files?`. `ohai` accepts multiple inputs and all but the first are printed (without styling or truncation) via `puts`. (I find it annoying that `ohai` truncates its first argument, but that's not something that can be solved here.)